website/docs/r/public_ip.html.markdown

```diff
@@ -42,7 +42,11 @@ The following arguments are supported:
 
 * `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
 
-* `public_ip_address_allocation` - (Required) Defines whether the IP address is stable or dynamic. Options are Static or Dynamic.
+* `sku` - (Optional) The SKU of the Public IP. Accepted values are `Basic` and `Standard`. Defaults to `Basic`.
+
+~> **Note** Public IP Standard SKUs require `public_ip_address_allocation` to be set to `static`.
```

Comment:
Should this be in a validation?

Reply:
@justaugustus apologies for the delayed response - we've been trying to get the SDKv12 migration finished up asap (which turned out to be a considerably larger job than we initially anticipated).

At the current time it's only possible to access the value of that specific field in the validation method on the Schema - as such I believe we'd need to do this in the `resourceArmPublicIpCreate` method?

Reply:
I've added a commit to explain that this feature is still in Preview and needs to be opted-in